src/lib.rs

```diff
@@ -197,6 +197,106 @@ impl Vessel {
+                    let mut single_cmd = Command::new(moc);
+                    if let Some(args) = moc_args {
+                        single_cmd.args(args.split(' '));
+                    }
+
+                    let entry_wasm_output = temp_dir.path().join(format!("output_{}.wasm", idx));
+                    single_cmd.args(["-o", entry_wasm_output.to_str().unwrap()]);
+
+                    // Add package dependencies
+                    for dep_package in &dependencies {
+                        let path = download_package(dep_package, false)?;
+                        single_cmd.arg("--package").arg(&dep_package.name).arg(path);
+                    }
+
+                    single_cmd.arg(&entry_point);
+
+                    info!("Running command: {:?}", single_cmd);
+                    let output = single_cmd
+                        .output()
+                        .context(format!("Failed to run {single_cmd:?}"))?;
+
+                    if !output.status.success() {
+                        return Err(anyhow::anyhow!(
+                            "Failed to verify \"{}\" with:\n{}",
+                            package.name,
+                            String::from_utf8(output.stderr)?
+                        ));
+                    }
+
+                    // Validate generated WASM file
+                    if entry_wasm_output.exists() {
+                        let validate_cmd = Command::new("wasm-validate")
+                            .arg(&entry_wasm_output)
+                            .output()
+                            .context("Failed to run wasm-validate - make sure it's installed")?;
+
+                        if !validate_cmd.status.success() {
+                            return Err(anyhow::anyhow!(
+                                "WASM validation failed for \"{}\" with:\n{}",
+                                package.name,
+                                String::from_utf8(validate_cmd.stderr)?
+                            ));
+                        }
+                    }
+                }
+
+                info!("Verified \"{}\"", package.name);
+                Ok(())
+            }
+        }
+    }
+
+    pub fn compile_all(&self, moc: &Path, moc_args: &Option<String>) -> Result<()> {
+        let mut errors: Vec<(Name, anyhow::Error)> = vec![];
+        for package in &self.package_set.topo_sorted() {
+            if !errors.iter().any(|(n, _)| package.dependencies.contains(n)) {
+                if let Err(err) = self.compile_package(moc, moc_args, &package.name) {
+                    errors.push((package.name.clone(), err))
+                }
+            }
+        }
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            let err = anyhow::anyhow!(
+                "Failed to compile: {:?}",
+                errors
+                    .iter()
+                    .map(|(n, _)| n.clone())
+                    .collect::<Vec<String>>()
```

Comment:
```suggestion
                    .map(|(ref n, _)| n)
                    .collect::<Vec<_>>()
```

Not really important here, but maybe it's a bit of Rust for your toolbox ;)

Reply:
Thanks! I've applied it to old `verify_all` function too